app/TracingStrategyContext.tsx

```diff
@@ -0,0 +1,44 @@
+import React, { createContext } from 'react';
+import { TracingStrategy } from './tracingStrategy';
+import { ExposureHistoryProvider } from './ExposureHistoryContext';
+
+interface TracingStrategyContextState {
+  name: string;
+}
+
+const defaultState = {
+  name: 'initializingTracingStrategy',
+};
+
+const TracingStrategyContext = createContext<TracingStrategyContextState>(
+  defaultState,
+);
+
+interface TracingStrategyProps {
+  children: JSX.Element;
+  strategy: TracingStrategy;
+}
+
+const TracingStrategyProvider = ({
+  children,
+  strategy,
+}: TracingStrategyProps): JSX.Element => {
+  const StrategyPermissionsProvider = strategy.permissionsProvider;
```

Comment:
Im less of a fan of passing in a whole react component here. Ideally we keep a better separation from the data model layer and the ui application layer.  I think we could instead provide a list of Permissions which provide was to interact with the device and provide that to a generic Permissions Provider.  It would make sense to do a pass at this in a future commit though.